src/NDItk/multigroup/Metadata.hpp

```diff
@@ -48,10 +51,16 @@ class Metadata {
    */
   bool isMetadataKey( const std::string& keyword ) const {
 
-    return ( keyword == "zaid" ) || ( keyword == "library_name" ) || ( keyword == "date_source" ) ||
-           ( keyword == "date_processed" ) || ( keyword == "awr" ) || ( keyword == "at_wgt" ) ||
-           ( keyword == "temp" ) || ( keyword == "sig_0" ) || ( keyword == "num_grps" ) ||
-           ( keyword == "num_reac" );
+    return ( keyword == this->zaid_.keyword() ) ||
+           ( keyword == this->library_name_.keyword() ) ||
+           ( keyword == this->source_date_.keyword() ) ||
+           ( keyword == this->process_date_.keyword() ) ||
+           ( keyword == this->awr_.keyword() ) ||
+           ( keyword == this->atomic_weight_.keyword() ) ||
+           ( keyword == this->temperature_.keyword() ) ||
+           ( keyword == this->dilution_.keyword() ) ||
+           ( keyword == this->reactions_.keyword() ||
+           ( keyword.find( this->primary_groups_.keyword() ) == 0 ) );
```

Comment:
I confused myself. Primary groups is only for the neutrons, right? Why is the find necessary?

Reply:
This picks up num_grps as a keyword but also num_grps_0, num_grps_1001, etc.

Reply:
Okay, I'm confused: primary_groups is either num_grps or num_grps_XXXX, but outgoing_groups is also num_grps_XXXX?

Reply:
I thought primary is only num_grps? and any num_grps_XXX are outgoing, so that's how I've been treating these. This particular line ```keyword.find( this->primary_groups_.keyword() ) == 0 )``` finds all these cases. The ```find``` part in this line returns the position of where the aprticular string is encountered, and since it has to be at the beginning we compare to 0.

Reply:
I think I understand. Thanks!